### PR TITLE
MM-69521: prevent Boards notification worker panic from malformed block properties

### PR DESCRIPTION
**Pull request by nang2049**
#### Summary
Hardens the notification subscription worker against malformed block payloads that could panic the worker goroutine and crash the Boards plugin process.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-69521

## Change Impact: Medium 🟡

**Regression Risk:** Changes are limited to Boards notification subscription handling and block property validation, but they alter runtime behavior in a background worker and expand validation strictness around `fields.properties`. The panic-recovery path is now covered by new unit tests, reducing risk of unhandled crashes, though malformed payload edge cases can still surface differences in behavior.

**QA Recommendation:** Light manual QA—run a short smoke test for Boards notifications with both well-formed and malformed block payloads, and confirm the worker remains healthy after induced callback panics.  
*Generated by CodeRabbitAI*

### CODE REVIEW

**Review by avasconcelos114 — APPROVED** [review]
LGTM!

File: server/model/block.go [hunk 1]
```diff
@@ -193,6 +194,12 @@ func (b *Block) baseValidations() error {
 		}
 	}
 
+	if propsIface, ok := b.Fields["properties"]; ok && propsIface != nil {
```
Comment (edgarbellot) [inline server/model/block.go]: nit: `"properties": null` bypasses the type check in `baseValidations()` because the guard condition `ok && propsIface != nil` short-circuits before the type assertion.

**Review by edgarbellot — APPROVED** [review]
Looking good 🚀 thank you!
